Bottom: 4708e0e756ad1dd598a75e5ce77692840118112f
Top:    157e6a7f961919a389353525a8a276e1ab693815
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-11 23:47:43 +0000

Remove Settings.init_done variable

This variable was previously used to allow action scripts run prior to
loading the GUI to avoid GUI startup after they call a "Quit" action.
The requirement seems to date back prior to the HID split, where the
"Quit" action would call gtk_main_quit(), an operation which is illegal
if the GUI main loop was never started.

During the HID split, the QuitApplication() function was changed to
call exit(0), so this dance isn't really necessary at all now.



---

diff --git a/src/global.h b/src/global.h
index 19f4454..80bea72 100644
--- a/src/global.h
+++ b/src/global.h
@@ -668,7 +668,6 @@ typedef struct			/* some resources... */
     /* connections is done */
     AutoPlace;			/* flag which says we should force placement of the
 				   windows on startup */
-  int init_done;
 }
 SettingType, *SettingTypePtr;
 
diff --git a/src/main.c b/src/main.c
index bb79991..162a0e0 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1021,12 +1021,6 @@ main (int argc, char *argv[])
 #ifdef HAVE_LIBSTROKE
   stroke_init ();
 #endif
-  /*
-   * Set this flag to zero.  Then if we have a startup
-   * action which includes Quit(), the flag will be set
-   * to -1 and we can avoid ever calling gtk_main();
-   */
-  Settings.init_done = 0;
 
   if (Settings.ScriptFilename)
     {
@@ -1040,37 +1034,33 @@ main (int argc, char *argv[])
       hid_parse_actions (Settings.ActionString, 0);
     }
 
-  if (Settings.init_done == 0)
-    {
-      Settings.init_done = 1;
 #if HAVE_DBUS
-      pcb_dbus_setup();
+  pcb_dbus_setup();
 #endif
 
-      EnableAutosave ();
+  EnableAutosave ();
 
 #ifdef DEBUG
-      printf ("Settings.LibraryCommandDir = \"%s\"\n",
-	      Settings.LibraryCommandDir);
-      printf ("Settings.FontPath          = \"%s\"\n", 
-	      Settings.FontPath);
-      printf ("Settings.ElementPath       = \"%s\"\n", 
-	      Settings.ElementPath);
-      printf ("Settings.LibraryPath       = \"%s\"\n", 
-	      Settings.LibraryPath);
-      printf ("Settings.LibraryTree       = \"%s\"\n", 
-	      Settings.LibraryTree);
-      printf ("Settings.MakeProgram = \"%s\"\n",
-	      UNKNOWN (Settings.MakeProgram));
-      printf ("Settings.GnetlistProgram = \"%s\"\n",
-	      UNKNOWN (Settings.GnetlistProgram));
+  printf ("Settings.LibraryCommandDir = \"%s\"\n",
+          Settings.LibraryCommandDir);
+  printf ("Settings.FontPath          = \"%s\"\n", 
+          Settings.FontPath);
+  printf ("Settings.ElementPath       = \"%s\"\n", 
+          Settings.ElementPath);
+  printf ("Settings.LibraryPath       = \"%s\"\n", 
+          Settings.LibraryPath);
+  printf ("Settings.LibraryTree       = \"%s\"\n", 
+          Settings.LibraryTree);
+  printf ("Settings.MakeProgram = \"%s\"\n",
+          UNKNOWN (Settings.MakeProgram));
+  printf ("Settings.GnetlistProgram = \"%s\"\n",
+          UNKNOWN (Settings.GnetlistProgram));
 #endif
 
-      gui->do_export (0);
+  gui->do_export (0);
 #if HAVE_DBUS
-      pcb_dbus_finish();
+  pcb_dbus_finish();
 #endif
-    }
 
   return (0);
 }
diff --git a/src/misc.c b/src/misc.c
index b675090..cca9e1b 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -879,16 +879,7 @@ QuitApplication (void)
   else
     DisableEmergencySave ();
 
-  /*
-   * if Settings.init_done is not > 0 then we haven't even called
-   * gtk_main() yet so gtk_main_quit() will give an error.  In
-   * this case just set the flag to -1 and we will exit instead
-   * of calling gtk_main()
-   */
-  if (Settings.init_done > 0)
-    exit (0);
-  else
-    Settings.init_done = -1;
+  exit (0);
 }
 
 /* ---------------------------------------------------------------------------
